workflow/workflow.go

```diff
@@ -353,6 +353,66 @@ func MutableSideEffect(ctx Context, id string, f func(ctx Context) interface{},
 // DefaultVersion is a version returned by GetVersion for code that wasn't versioned before
 const DefaultVersion Version = internal.DefaultVersion
 
+// ExecuteWithVersion forces a specific version to be returned when GetVersion is executed for the first time,
+// instead of returning maxSupported version. This option can be used when you want to separate the versioning of the workflow code and
+// activation of the new logic in the workflow code, to ensure that your changes can be safely rolled back, if needed.
+//
+// For example, initially a workflow has the following code:
+//
+//	err = workflow.ExecuteActivity(ctx, foo).Get(ctx, nil)
+//
+// It should be updated to:
+//
+//	err = workflow.ExecuteActivity(ctx, bar).Get(ctx, nil)
+//
+// Following the steps below, your changes will be forward and backward compatible, keeping possible a safe rollback
+// to the previous version of the workflow code:
+//
+// 1. Keep execution of foo activity, add a support of bar activity
+//
+//	v := GetVersion(ctx, "fooChange", DefaultVersion, 1,  ExecuteWithVersion(DefaultVersion))
+//	if v == DefaultVersion {
+//		err = workflow.ExecuteActivity(ctx, foo).Get(ctx, nil)
+//	} else {
+//		err = workflow.ExecuteActivity(ctx, bar).Get(ctx, nil)
+//	}
+//
+// The code above supports replaying of workflow execution with both versions DefaultVersion and 1.
+// All new workflow executions will execute foo activity, because GetVersion with ExecuteWithMinVersion option
+// returns DefaultVersion, that will be recorded into the workflow history.
```

Comment:
```suggestion
// All new workflow executions will execute foo activity, because ExecuteWithVersion(DefaultVersion)
// causes GetVersion to return DefaultVersion rather than 1.
// In this example, this is also exactly the same as using ExecuteWithMinVersion().
```
I'm kinda divided on keeping/removing "will be recorded into the workflow history" here... for non-default versions it's true, but for this specific example it won't record the version at all because it's the default.  though the behavior will otherwise act as if it was recorded.